### DISCUSSION TITLE
How to use some field pro in List Operation.

### DISCUSSION THREAD

**Discussion by tatvn** [Laravel-Backpack/community-forum#767] [Q&A (Help)] [answered]
I use select2_from_array (pro version) in Create Operation then it work. But, when I use in List Operation then it not work. It understand as Select_from_array. How to fix it?
In Controller:
```php
CRUD::addField([
                'name' => 'select2',
                'type'    => 'select2_from_array',
                'options'   => [
                       1 => 'a'
                       2 => 'b'
                       3 => 'c'
                ],
                'default'     => '',
                'wrapper' => ['class' => 'form-group col-md-1'],
]);
```
In Blade:
```blade
<div class="card">
        <div class="card-body row">
            @include('crud::inc.show_fields', ['fields' => $crud->fields()])
        </div>
</div>
```

**Comment by karandatwani92** [comment]
Hey @THAIANH98 

- In Edit & Create Operations we add **[fields](https://backpackforlaravel.com/docs/6.x/crud-fields)** `CRUD:: addField([]);`

- In List/Show Operations we add **[columns](https://backpackforlaravel.com/docs/6.x/crud-columns)**`CRUD::addColumn([]);`

In case you want editable columns inside the list operation. This would require our [editable-columns](https://backpackforlaravel.com/products/editable-columns) package.


Let me know if I'm missing something. Happy Coding!

**Comment by tatvn** [comment]
Hi @karandatwani92, I understand what you say. But I wrote this code in custom.blade.php
```blade
<div class="card">
        <div class="card-body row">
            @include('crud::inc.show_fields', ['fields' => $crud->fields()])
        </div>
</div>
```
I retrieve the data using this code
```php
View::composer('admin.project.form_search', function ($view) {
          CRUD::addField([
                'name' => 'select2',
                'type'    => 'select2_from_array',
                'options'   => [
                       1 => 'a'
                       2 => 'b'
                       3 => 'c'
                ],
                'default'     => '',
                'wrapper' => ['class' => 'form-group col-md-1'],
         ]);
         $view->with('crud', $this->crud);
         Widget::add()->to('before_content')->type('div')->class('row')->content([
            [
                'type'     => 'view',
                'view'     => 'custom',
            ]
        ]);
});
```
and it understands that it is select_from_array. How to it understands that it is select2_from_array? Thanks

**Comment by karandatwani92** [comment, marked as answer]
OK @THAIANH98

are you trying to use CRUD fields in the list operation via the Widget section(before_content)?

Let's assume somehow we did that. but how you are assuming they will work?
It will require a form, an action. a URL to submit or an Ajax setup, further validation - error messages, etc.

I believe, using CRUD fields here can't get all this in list operation. (Fields are not written to work inside list operation)

 
### Now let's assume you have something else in your mind!
Now let's focus back on your question.

The problem must be the missing selec2/CSS resources. 
`@basset('https://unpkg.com/select2@4.0.13/dist/css/select2.min.css')`
`@basset('https://unpkg.com/select2-bootstrap-theme@0.1.0-beta.10/dist/select2-bootstrap.min.css')`
`@basset('https://unpkg.com/select2@4.0.13/dist/js/select2.full.min.js')`

**Why it's not loading the resources?**
`crud_fields_scripts` & `crud_fields_styles` are pushed on only on create & edit operation via `vendor/backpack/crud/src/resources/views/crud/form_content.blade.php`.

Where as list operation uses `@stack('crud_list_scripts')` & `@stack('crud_list_styles')`

Thus, you have to include resources by yourself.

I hope this helps to understand better.

**Reply by tatvn** [reply]
how to include list operation? Please help me.

**Reply by karandatwani92** [reply]
You can add CSS/JS to particular list operation using Script & Style widgets, check [here](https://backpackforlaravel.com/docs/6.x/base-widgets#script-1)
